scene_graph_benchmark/relation_head/sparse_targets.py

```diff
@@ -79,8 +80,10 @@ def _get_tensor_from_boxlist(proposals, field='labels'):
             im_inds = torch.cat(
                 (im_inds, im_ind * torch.ones(num_proposals_im, 1)), dim=0)
 
-    # TODO: support both cpu and gpu
-    im_inds_batch = torch.Tensor(im_inds).long().cuda()
+    im_inds_batch = torch.Tensor(im_inds).long()
+    if framework_device == "cuda":
+        im_inds_batch.to(torch.device("cuda"))
```

Comment:
are you sure this is how to use the `to()` method?
My understanding is that you have to write:

```
im_inds_batch = im_inds_batch.to(torch.device("cuda"))
```

Reply:
It is not necessary. From the documentation example (https://pytorch.org/docs/stable/generated/torch.Tensor.to.html):

`>>> tensor = torch.randn(2, 2)  # Initially dtype=float32, device=cpu
>>> tensor.to(torch.float64)
tensor([[-0.5044,  0.0005],
        [ 0.3310, -0.0584]], dtype=torch.float64)

>>> cuda0 = torch.device('cuda:0')
>>> tensor.to(cuda0)
tensor([[-0.5044,  0.0005],
        [ 0.3310, -0.0584]], device='cuda:0')`

Anyway, I tried on the software to be sure. Here is the modification I have made on the `sparse_targets.py` file:

<img width="403" alt="image" src="https://user-images.githubusercontent.com/72139634/161518168-a62c0b54-f95d-4e24-897c-f5e92b57da7d.png">

and here is the output:

<img width="491" alt="image" src="https://user-images.githubusercontent.com/72139634/161518438-6753fc8c-c026-44c9-b322-3b985e58371b.png">

Reply:
The example is misleading. because the pint out is the print of the return value.
You test needa to be run on a machine with both gpu and cpu and you should do:
```
tensor_original = torch.randn(2, 2)  # Initially dtype=float32, device=cpu

cuda0 = torch.device('cuda:0')


tensor_returned = tensor_original.to(cuda0)

print (tensor_original)
print (tensor_returned)


```

Reply:
You are right, I am correcting it.